transport/src/main/java/io/netty/channel/DefaultSlicedFileRegion.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * A derived {@link FileRegion} which shares its parent's {@link FileChannel}.
+ */
+public class DefaultSlicedFileRegion implements FileRegionHelper.InternalFileRegion {
```

Comment:
@Scottmitch should we make `InternalFileRegion` a top-level class? Seems strange implementing an inner class of `FileRegionHelper`.